py/frequenz/client/weather/_historical_forecast_iterator.py

```diff
@@ -65,29 +65,26 @@ async def __anext__(self) -> weather_pb2.LocationForecast:
         Raises:
             StopAsyncIteration: If there are no more historical weather forecasts.
         """
-        if len(self.location_forecasts) == 0 and self.page_token == EMPTY_PAGE_TOKEN:
+        if self.page_token == EMPTY_PAGE_TOKEN:
             raise StopAsyncIteration
 
-        if self.location_forecasts is None or len(self.location_forecasts) == 0:
-            pagination_params = pagination_params_pb2.PaginationParams()
-            pagination_params.page_size = PAGE_SIZE
-            if self.page_token is not None:
-                pagination_params.page_token = self.page_token
-
-            response: Any = (
-                await self._stub.GetHistoricalWeatherForecast(  # type:ignore
-                    weather_pb2.GetHistoricalWeatherForecastRequest(
-                        locations=(location.to_pb() for location in self.locations),
-                        features=(feature.value for feature in self.features),
-                        start_ts=self.start_ts,
-                        end_ts=self.end_ts,
-                        pagination_params=pagination_params,
-                    )
-                )
+        pagination_params = pagination_params_pb2.PaginationParams()
+        pagination_params.page_size = PAGE_SIZE
```

Comment:
I agree this should be part of the configuration. In PR I kept it as a constant to avoid the additional complexity. There is no configuration for the client now, so we would need to introduce the whole class/module for this rather than just add some value to an existing file.

So I suggest this should be attended to in another PR. I created an issue for this: https://github.com/frequenz-floss/frequenz-api-weather/issues/96

Reply:
I think he just meant take it as a parameter in the method.  Configurable doesn't mean from a file.

Reply:
Probably in the `__init__`, and you could give it a default value.

Reply:
Ok, taking your idea with setting the value in __init__ and using a default value.

Reply:
>So I suggest this should be attended to in another PR. I created an issue for this: https://github.com/frequenz-floss/frequenz-api-weather/issues/96

Yes, I mean an parameter in the `__init__` method, sorry if that wasn't clear enough. I don't think we need that issue.